# Consistencia de datos

De acuerdo al archivo `diccionarioVariables.xlsx`, se debe examinar la consistencia de datos de la variable renombrada `hijos` descrita como "Número de hijos dependientes".

In [1]:
import pandas as pd

df = pd.read_pickle("../data/interim/df_tipoDato.pkl")

In [2]:
df['hijos'].value_counts()

hijos
0.000000    10905
1.000000     8361
2.000000     2779
3.000000      683
4.000000      127
0.738639       24
5.000000       17
7.000000        5
0.710668        3
6.000000        3
Name: count, dtype: int64

Los valores inconsistentes de esta variable son: `0.738639` y `0.710668`. No tiene sentido que un operario tenga como número de hijos dependientes estas cantidades.

Así que la correción será crear una variable dummy `al_menos_un_hijo` cuyos valores serán 1 si el operario tiene un hijo y 0 si no lo tiene.

In [3]:
df['al_menos_un_hijo'] = df['hijos'].apply(lambda x: 0 if x == 0 else 1)
df['al_menos_un_hijo'].value_counts()

al_menos_un_hijo
1    12002
0    10905
Name: count, dtype: int64

Eliminamos la variable `hijos`:

In [4]:
df_datosConsistentes = df.drop('hijos', axis=1)

Guardando base de datos:

In [5]:
df_datosConsistentes.to_pickle("../data/interim/df_datosConsistentes.pkl")